src/test/lib/concurrency/stress_test.cpp

```diff
@@ -361,9 +364,12 @@ TEST_F(StressTest, NodeQueueSchedulerSemaphoreIncrements) {
 // Similar to test above, but here we make tasks dependent of each other which means only non-dependent tasks will be
 // scheduled.
 TEST_F(StressTest, NodeQueueSchedulerSemaphoreIncrementsDependentTasks) {
-  constexpr auto DEPENDENT_JOB_TASKS_LENGTH = size_t{10};
+  constexpr auto DEPENDENT_JOB_TASKS_LENGTH = uint32_t{10};
   constexpr auto SLEEP_TIME = std::chrono::milliseconds{1};
-  const auto job_count = CORES_PER_NODE * 32;
+
+  // Ensure there is at least one job left after each worker pulled one.
+  const auto min_job_count = DEPENDENT_JOB_TASKS_LENGTH * CPU_COUNT + 1;
+  const auto job_count = std::max(min_job_count, CPU_COUNT * 8 * (HYRISE_WITH_TSAN ? 1 : SMALL_SANITIZER_LOAD_FACTOR));
```

Comment:
Use the sanitizer load factor whenever we don't run the sanitizer?

Reply:
Also not super happy with the naming. Better ideas? I wanted to avoid having the factor as 1/4 or 1/10 and add a lot of casting. `[SMALL|MEDIUM]_LOAD_FACTOR`?